src/Components/Web/src/Image/Image.cs

```diff
@@ -0,0 +1,267 @@
+
+    private bool IsInteractive => _renderHandle.IsInitialized &&
+                                _renderHandle.RendererInfo.IsInteractive;
+
+    [DisallowNull] private ElementReference? Element { get; set; }
+
+    [Inject] private IJSRuntime JSRuntime { get; set; } = default!;
+
+    [Inject] private ILogger<Image> Logger { get; set; } = default!;
+
+    /// <summary>
+    /// Gets or sets the source for the image.
+    /// </summary>
+    [Parameter] public ImageSource? Source { get; set; }
+
+    /// <summary>
+    /// Gets or sets the attributes for the image.
+    /// </summary>
+    [Parameter(CaptureUnmatchedValues = true)] public Dictionary<string, object>? AdditionalAttributes { get; set; }
+
+    void IComponent.Attach(RenderHandle renderHandle)
+    {
+        if (_renderHandle.IsInitialized)
+        {
+            throw new InvalidOperationException("Component is already attached to a render handle.");
+        }
+        _renderHandle = renderHandle;
+    }
+
+    Task IComponent.SetParametersAsync(ParameterView parameters)
+    {
+        var previousSource = Source;
+
+        // Set component parameters
+        parameters.SetParameterProperties(this);
+
+        // Initialize on first parameters set
+        if (!_initialized)
+        {
+            Render();
+            _initialized = true;
+            return Task.CompletedTask;
+        }
+
+        if (Source != null && !string.Equals(previousSource?.CacheKey, Source.CacheKey, StringComparison.Ordinal))
+        {
+            Render();
+        }
+
+        return Task.CompletedTask;
+    }
+
+    async Task IHandleAfterRender.OnAfterRenderAsync()
+    {
+        if (!IsInteractive || Source == null)
+        {
+            return;
+        }
+
+        if (_currentSource != null && string.Equals(_currentSource.CacheKey, Source.CacheKey, StringComparison.Ordinal))
+        {
+            return;
+        }
+
+        // Cancel any in-progress load operation
+        try { _loadCts?.Cancel(); } catch { }
+        _loadCts?.Dispose();
+        _loadCts = new CancellationTokenSource();
+        var token = _loadCts.Token;
+
+        _currentSource = Source;
+
+        try
+        {
+            await LoadImage(Source, token);
+        }
+        catch (OperationCanceledException)
+        {
+
+        }
+    }
+
+    private void Render()
+    {
+        if (!_hasPendingRender && _renderHandle.IsInitialized)
+        {
+            _hasPendingRender = true;
+            _renderHandle.Render(BuildRenderTree);
+            _hasPendingRender = false;
+        }
+    }
+
+    private void BuildRenderTree(RenderTreeBuilder builder)
+    {
+        builder.OpenElement(0, "img");
+
+        if (!string.IsNullOrEmpty(_currentObjectUrl))
+        {
+            builder.AddAttribute(1, "src", _currentObjectUrl);
+        }
+
+        builder.AddAttribute(2, "data-blazor-image", "");
```

Comment:
At some point we will want to change some of the naming here. `data-inline-image` or `data-dynamic-image` could be good options. (Let's keep it like this for now).

In general, we don't want to have `blazor` showing everywhere